app/auth/views.py

```diff
@@ -0,0 +1,157 @@
+from . import auth_blueprint
+from flask.views import MethodView
+from flask import make_response, request, jsonify
+from app.models import User, BlackListToken
+from flasgger import swag_from
+
+
+def get_authenticated_user(request):
+    auth_header = request.headers.get('Authorization')
+    if not auth_header:
+        return None
+    try:
+        access_token = auth_header
+    except (IndexError, ValueError):
+        return 'Authorization header is in wrong format.'
+    if not access_token:
+        return None
+    else:
+        user_id = User.decode_token(access_token)
+        if not isinstance(user_id, str):
+            # user is authenticated so get the user
+            user = User.query.get(user_id)
+            return user
+        elif user_id == 'You are already logged out':
+            return user_id
+        else:
+            return None
+
+
+#  class to register new user
+class RegistrationView(MethodView):
+    """document api"""
+
+    @swag_from('swagger_docs/register_user.yaml', methods=['POST'])
+    def post(self):
+        # Handle POST request for this view. Url ---> /auth/register"""
+        # Query to see if the user already exists
+        post_data = request.get_json(force=True)
+        # check if posted username is already in the database
+        user = User.query.filter_by(username=post_data['username']).first()
+        if not user:
+            # There is no user so we'll try to register new user
+            try:
+                # Register new user
+                username = post_data['username']
+                fullname = post_data['fullname']
+                password = post_data['password']
```

Comment:
@code-sleuth as per the concern raised by @dennisja  it would be better to use `username =post_data.get('username')` and `password = post_data.get('password')`